jlloyd0430/Yarn.solutions
# Commits Allowing Users to Link Wallets from their Profile Page

commit 246f6679b1d3cc2db0a370d82f45e8f65513d8e7
Author: Thaddeus Diamond <thaddeus.diamond@gmail.com>
Date:   Tue Aug 1 18:37:29 2023 -0500

    Wallet Integration: Include basic wallet connector for user.wallets
    
    We add a new field `wallets` to our user's schema.  We intend to have an
    authenticated endpoint(s) that can add/remove to this array for the
    current user.
    
    However, this change simply modifies the profile view and edit pages to:
    1. Show any wallets (currently blank)
    2. Add a wallet connector
    
    NOTE: This requires us to load the `NftToolkit` from CDN as we rely on
    the WildTangz dApp connector.
    
    NOTE: There is an unrelated change for Mac OS X development (.DS_Store)
    
    [ Documentation: None required (no setup changes) ]
    [ Testing: npm run devStart ]

commit 81fd45c5f235d27fcb8fce72fcfbd0d89e0d6423
Author: Thaddeus Diamond <thaddeus.diamond@gmail.com>
Date:   Tue Aug 1 22:30:55 2023 -0500

    Wallet Connector: Add backend for adding a wallet stake key
    
    This uses CIP-8 (https://cips.cardano.org/cips/cip8/) to verify the
    signature of the current Epoch's first block time (in seconds) before
    adding the user's wallet.  The algorithm is:
    1. Wait for `CARDANO_DAPP_JS_CONNECT` message on the client side (user
       successfully connected)
    2. Retrieve the user's stake key and prompt for a signature on it
    3. Send the signed message of the current Epoch's first block time to
       the backend and include credentials
    
    On the backend, we simply:
    1. Retrieve the current user from the session
    2. Check if this is a dupe-linked stake key to that user
    3. Use MongoDB to $push the new ID in
    
    [ Testing: npm run devStart ]

commit e99eb29623e35677cb48577c82bb6f98f2b17d8c (HEAD -> thad-dev, fork/thad-dev, main)
Author: Thaddeus Diamond <thaddeus.diamond@gmail.com>
Date:   Tue Aug 1 23:42:06 2023 -0500

    Wallet Connector: /remove_wallet endpoint for remove wallet
    
    This commit adds a DELETE endpoint for the user that allows them to
    remove a wallet.  The algorithm on the backend is simple:
    1. Get the user from the current session
    2. Check that the wallet is actually linked
    3. Pull it from MongoDB using the `$pull` operator
    
    Then on the frontend, we add JavaScript and styling to make rows with
    the existing linked wallets and a trash icon next to each.
    
    NOTE: We move the wallet loading to JavaScript to encourage code reuse,
    though when a new wallet is successfully linked, though it's not
    strictly necessary.
    
    [ Testing: npm run devStart ]

commit 3d9d1af2cee74caab1c128d8e4a5203f242cd4ea (HEAD -> thad-dev, fork/thad-dev)
Author: Thaddeus Diamond <thaddeus.diamond@gmail.com>
Date:   Tue Aug 1 23:48:19 2023 -0500

    Remove WalletConnect for easier debugging/faster loading
